agent/taskresource/cgroup/cgroup.go

```diff
@@ -44,143 +44,231 @@ var (
 
 // CgroupResource represents Cgroup resource
 type CgroupResource struct {
-	taskARN             string
-	control             cgroup.Control
-	cgroupRoot          string
-	cgroupMountPath     string
-	resourceSpec        specs.LinuxResources
-	ioutil              ioutilwrapper.IOUtil
-	createdAt           time.Time
-	desiredStatusUnsafe taskresource.ResourceStatus
-	knownStatusUnsafe   taskresource.ResourceStatus
+	taskARN                            string
+	control                            cgroupres.Control
+	cgroupRoot                         string
+	cgroupMountPath                    string
+	resourceSpec                       specs.LinuxResources
+	IOUtil                             ioutilwrapper.IOUtil
```

Comment:
Why is this changed to be public?

Reply:
so that we could use it in tests [here](https://github.com/aws/amazon-ecs-agent/pull/1332/files#diff-af65229648e7812b105ca712053403b3R152).

Reply:
Then can you add a setter for it?

Reply:
setter is a good idea, though right now it's used only in the tests. added the same.